### PR TITLE
Updating typings for latest changes to fuzzaldrin-plus

### PR DESCRIPTION
**Pull request by mdahamiwal**
case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url https://github.com/jeancroy/fuzzaldrin-plus/tree/master/src
- [ ] it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

**Comment by mdahamiwal** [comment]
@vvakame , can you add someone to review it, Thanks :)

**Comment by mdahamiwal** [comment]
@andy-ms , can you review this change? Thanks

File: fuzzaldrin-plus/index.d.ts [hunk 1]
```diff
@@ -5,25 +5,32 @@
 
 export as namespace fuzzaldrin;
 
-export interface IQueryOptions {
+export interface QueryOptions {
```
Comment (JKillian) [inline fuzzaldrin-plus/index.d.ts]: I understand that the `I` prefix might not be needed, but I'd recommend not changing this, simply for the sake of backward compatibility. No reason to make users using these typings have to change their code.

**Review by jkillian — CHANGES_REQUESTED** [review]
I also recommend not merging this until fuzzaldrin-plus publishes a new release. When it is released, you'll have to change a version number as part of this PR to match the newly released version.

File: fuzzaldrin-plus/index.d.ts (hunk 1 above)
Comment (mdahamiwal) [inline fuzzaldrin-plus/index.d.ts]: @JKillian, I understand your point but my understanding is that _types-2.0_ branch is meant for future typings (that they would be merged to master) and only the ones in _master_ is available to TSD. Is there a distinction here on when should we put something in _master_ vs other branches in this repository? Are Devs taking dependency on typings available on non-master branches?

Comment (minestarks) [inline fuzzaldrin-plus/index.d.ts]: @mdahamiwal that's not quite accurate. types-2.0 types are published to @types and are meant to be used by the TypeScript 2.0 compiler. So unfortunately yes, you do need to worry about backwards compatibility. See the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md) for clarification.

**Comment by minestarks** [comment]
@JKillian @mdahamiwal This PR is rather stale and sounds like there is some unresolved discussion. Should it be merged? If no further comments I plan to close it out in a couple of days.

**Comment by JKillian** [comment]
Well, it shouldn't be merged yet, because this reflects new changes that aren't in a released version of `fuzzaldrin-plus` yet. Once that library releases a new version, then it'd be time to consider this in more detail

**Comment by mdahamiwal** [comment]
@minestarks, we just released a new version of the lib that matches the typing changes in this PR. So we should be Ok to merge this. Is there a way to keep older version typings as well ? I see some details in readme to name older version like <lib>-<version>.d.ts, will that be taken care by the typing publisher? How do we do it for other libraries that are widely used like Jquery for instance?

**Comment by andy-ms** [comment]
Why not directly [bundle](http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html)  types with your package? That way you don't have to worry about keeping DefinitelyTyped in sync, and users don't have to install two different packages.
As for publishing multiple versions, this isn't yet a feature that we support.

**Comment by mdahamiwal** [comment]
Thanks, that would make more sense, will move d.ts to actual repository.I am fine with abandoning this PR.

**Comment by andy-ms** [comment]
Best of luck. Be sure to notify us when you've done so, so we can deprecate `@types/fuzzaldrin-plus`.

**Pull request closed without merging**